src/android/toga_android/app.py

```diff
@@ -92,3 +112,19 @@ def set_on_exit(self, value):
 
     def add_background_task(self, handler):
         self.loop.call_soon(wrapped_handler(self, handler), self)
+
+    async def invoke_intent_for_result(self, intent):
+        """
+        Calls an Intent and waits for its result
+
+        :param Intent intent: The Intent to call
+        :returns: A Dictionary containing "resultCode" (int) and "resultData" (Intent or None)
+        :rtype: dict
+        """
+        self._listener.last_intent_requestcode += 1
+        code = self._listener.last_intent_requestcode
+        result_future = asyncio.Future()
+        self._listener.running_intents[str(code)] = result_future
```

Comment:
Why the conversion to str for the dictionary key?

Reply:
Are keys not required to be strings in Python dictionaries?

Reply:
No, I just found that keys can be of any type...I'm still a Python newbie, I'm afraid..
I'll use the int values as keys, then

Reply:
Minor clarification - they dict keys can be any *hashable* type. You can't use a dictionary as a hash key, for example, because the dict itself can change, so you can't generate a consistent hash. But integers, strings, floats, most simple objects, and tuples of those primitives are all fair game as dict keys.